src/protocol/composite.rs

```diff
@@ -149,7 +149,8 @@ impl QueryVersionRequest {
         validate_request_pieces(header, value, None, Some(QUERY_VERSION_REQUEST))?;
         let (client_major_version, remaining) = u32::try_parse(value)?;
         let (client_minor_version, remaining) = u32::try_parse(remaining)?;
-        let remaining = &remaining[..(4 - (remaining.len() % 4)) % 4];
+        let remaining = remaining.get(..(4 - (remaining.len() % 4)) % 4)
+            .ok_or(ParseError::ParseError)?;
```

Comment:
This goes completely away when/if `check_exhausted` is removed, right? If not, I will have to think again about what this actually does.

Reply:
It would, yes. That's just to consume enough bytes to round us up to a multiple of 4.

Reply:
But this assumes that the length of the slice was a multiple of 4 to begin with, right? If so, I now understand what this does. "Remaining length should be a multiple of 4" is then equivalent to "we parsed a multiple of 4".

Reply:
No longer relevant since we're not validating lengths anymore.